code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm

```diff
@@ -1142,3 +1142,19 @@
 			drinker.adjust_hallucinations(60 SECONDS * REM * seconds_per_tick)
 
 	return ..()
+
+/datum/reagent/consumable/t_letter
+	name = "T"
+	description = "You expected to find this in a soup, but this is fine too."
+	color = "#583d09" // rgb: 88, 61, 9
+	taste_description = "one of your 26 favorite letters"
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/consumable/t_letter/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
+	if(ishuman(affected_mob) && HAS_MIND_TRAIT(affected_mob, TRAIT_MIMING))
+		affected_mob.set_silence_if_lower(MIMEDRINK_SILENCE_DURATION)
+		affected_mob.adjust_drowsiness(-6 SECONDS * REM * seconds_per_tick)
+		affected_mob.AdjustSleeping(-40 * REM * seconds_per_tick)
+		if(affected_mob.getToxLoss() && SPT_PROB(10, seconds_per_tick))
+			affected_mob.adjustToxLoss(-4, FALSE, required_biotype = affected_biotype)
+	return ..()
```

Comment:
invert this to be an early return `HAS_MIND_TRAIT` so something like
```dm
if(!ishuman(affected_mob) || !HAS_MIND_TRAIT(affected_mob, TRAIT_MIMING))
    return ..()

// actual working code goes here
```

Reply:
oops, right